lib/Sema/TypeCheckDecl.cpp

```diff
@@ -2603,6 +2603,10 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
     if (VD->getAttrs().hasAttribute<DynamicReplacementAttr>())
       TC.checkDynamicReplacementAttribute(VD);
+
+    // Now check all the accessors.
+    for (auto *accessor : VD->getAllAccessors())
+      visit(accessor);
```

Comment:
Same comment about `getAllAccessors`, although it seems even more pertinent here because of the order-sensitivity of which accessors will exist at this point.

Reply:
Yeah, eventually typeCheckDecl() will only be called on parsed accessors.